ASWF/meetings/tsc/2020-11-09.md

```diff
@@ -0,0 +1,251 @@
+        - Carol: Seems like added complication, but could look at it as 
+          building block of step in direction of where we want to be. More 
+          educated users knowing what they are doing in color managed scenario. 
+          Until we get to that point, could view this as interim. Helps to 
+          teach and educate. Like in ACES gamut mapping wg, all agree it's not 
+          ideal, but we have to do what helps.
+        - Mark B: Having multiple named transforms to reconcile error from 
+          another user. Are you anticipating a user assigning multiple named 
+          transforms in Maya UI, or still picking single transforms from named 
+          lists. If the later, still at mercy of config author, at which point 
+          it's just a doc or training session to explain choices. 
+        - Doug: Tried providing combinations in SynColor, many options, huge 
+          list of matrix and log-to-lin combinations and nightmare to deal 
+          with. Not desireable way of solving the problem.
+        - J: Does TSC have a point of view on what color space interaction and 
+          user experience should be.
+        - Michael: Think that's an important topic for discussion. Created 
+          issue #1174 for UX discussion where some of this could be determined.
+        - Kevin: What I really need is a Nuke node, op builder, to put ops in 
+          a row.
+        - Michael: Suggested a GSoC project to create OCIO GUI last year for 
+          exactly that reason. Would be awesome to have visual method for 
+          authoring configs.
+            - Let's bookmark this conversation for next week. Gives us time to 
+              think this through more and can follow up to make an agreement.
+
+* CUDA support:
+    - Doug: Larry brought up topic about CUDA support, pressing issue for 
+      OSL and a number of CG rednderers. 
+    - Michael: We should follow up with Dennis about his OpenCL renderer. Could 
+      perhaps influence other GPU compute implementations like Metal, Vulkan, 
+      etc.
+    - Mark B: Think this need is dying, since we now have OpenGL compute 
+      shader.
+    - Michael: CUDA requirement might be tied to OptiX. Could also explore 
+      tools to emit to multiple APIs. Spir-V.
+    - Mark: Spir-V builder needs large dependency graph, needs LLVM. Not stable 
+      dependencies.
+    - Patrick: Something to keep in mind. A lot of effort in current impl to 
+      not have direct OpenGL dependency in core. Clear distinction between core 
+      and shader language. Try to keep that. Advantage of OCIO.
+    - Michael: Could be like OIIO which has plugins depending on optional 
+      dependencies. More ideal to only depend on runtime though.
+    - Mark B: Windows has HLSL, Mac has Metal. Don't want to support all that 
+      stuff in the GPU impl. Emitting HLSL on Linux, etc. Could have call to 
+      get native shader code.
```

Comment:
I'm sorry, I should really have been there.

This is wholly unrelated to "viewport graphics" APIs like OpenGL or Vulkan.

High-end rendering is becoming rapidly focused on renderers written largely in Cuda (and mostly using OptiX for the actual ray tracing bits).

Reply:
@lgritz since I know many renders are more passive when it comes to color management (i.e. textures and lights are already in rendering space, and rendered results match that), is the need to also handle texture color management on the GPU in the renderer? Or is the need to fully support OSL, etc. in CUDA  to maintain OSL's OCIO support?

Reply:
@michdolan While it is best practices to preconvert your textures, many renderers let you hang yourself and have OCIO colorspace transform nodes in a shader graph.  Maya has this natively in their file nodes too.

https://docs.arnoldrenderer.com/display/A5AFMUG/ACES+Workflow.
https://docs.chaosgroup.com/display/VRAYHOUDINI/V-Ray+OCIO
https://docs.chaosgroup.com/display/VMAYA/VRayTexOCIO

Reply:
There are other reasons, also, why people might want to do color space conversions inside a shader, on individual colors. Mostly what I'm asking about is not wholesale conversion of an image on the way in (texture, can use CPU as it reads) or on the way out (can be CPU if it's writing to a file, or one of the "viewport graphics" APIs if it's live to a display).